consumer.go

```diff
@@ -376,11 +393,13 @@ func (child *PartitionConsumer) Close() error {
 	// close itself)
 	close(child.dying)
 
+	for _ = range child.messages {
```

Comment:
as in the producer's shutdown draining of successes, I believe this must be in a goroutine to prevent deadlock